### ISSUE TITLE
Help users share masked trace data for debugging

### ISSUE THREAD

**Issue by shakuzen** [openzipkin/zipkin#2426]
**Feature:**
I want to leave the details of how this is accomplished to discussion, but the idea is that as a user, you would be able to generate and share with Zipkin maintainers a masked copy of your tracing data that still reproduces an issue so that maintainers can immediately reproduce and debug the issue.

In short, some processing that transforms the original trace data to masked trace data in a secure way.

**Discussion questions:**
* What in tracing data is generally considered sensitive?
* What transformations can be done that will not affect reproducibility of the original issue?
* UX: how should this feature be presented? (e.g. button in the UI, separate endpoint, script)

**Rational**
Most Zipkin sites cannot share their tracing data publicly as-is, but having tracing data that reproduces issues experienced by users will enable debugging of issues to happen much faster and more accurately.

**Example Scenario**
A Zipkin user finds what appears to be a bug in the way a trace is displayed and wants to report it without leaking any sensitive data that might be in the trace. With a way to share this quickly, the user doesn't need to make and send masked screenshots or written descriptions that usually result in a lot of back-and-forth communication before common understanding of the problem is reached.

**Prior Art**
Not aware of any prior art in this area but will update this section if some is brought to light.

**Bonus**
This would make it easier for us to collect additional interesting trace data from sites to use in our test data.

**Comment by huydx** [comment]
Idea from me: a dropdown button with 2 options: Download and Download With Masking 
Download with Masking will simply mask data with random string (except trace id etc) for display purpose debugging. The logic will be implemented in UI side.

**Comment by codefromthecrypt** [comment]
my 2p. masking with random will make certain troubleshooting cases really
difficult especially service names.

I would recommend some other substitution approach, such as Latin words,
for service names. and honor the same shapes for things like http URLs. (ex
same depth and exact cardinality).

things that likely should stay the same are timestamps, duration, span
kind, ports (while replacing IPs consistently)

I'm basically describing what I do manually.

**Comment by codefromthecrypt** [comment]
also I think masking is just one filter. you can consider PII redaction as
an independent step. ex masking of phone number or account data is good for
masking.

whatever we do, we should alert the user they are responsible to edit the
file manually before posting anywhere. for example, our masking can take
some heavy lifting out of the process but it is still their responsibility
to not leak data not the responsibility of our UI code.

**Comment by shakuzen** [comment]
> whatever we do, we should alert the user they are responsible to edit the file manually before posting anywhere. for example, our masking can take some heavy lifting out of the process but it is still their responsibility to not leak data not the responsibility of our UI code.

I fully agree. We want to make this clearly stated however the feature surfaces.
